titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/reverseproxy/http/EurekaReactorHttpClientFactory.java

```diff
@@ -0,0 +1,73 @@
+package com.netflix.titus.ext.eureka.reverseproxy.http;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.EurekaClient;
+import com.netflix.titus.common.network.reverseproxy.http.ReactorHttpClientFactory;
+import com.netflix.titus.common.util.CollectionsExt;
+import com.netflix.titus.ext.eureka.reverseproxy.EurekaReverseProxyConfiguration;
+import reactor.ipc.netty.http.client.HttpClient;
+
+/**
+ * FIXME {@link HttpClient} accepts single address only. We need a wrapper that would work with Eureka based name resolver.
+ */
+@Singleton
+public class EurekaReactorHttpClientFactory implements ReactorHttpClientFactory {
+
+    private final EurekaClient eurekaClient;
+    private final EurekaReverseProxyConfiguration configuration;
+    private final ConcurrentMap<String, HttpClient> clients = new ConcurrentHashMap<>();
+
+    @Inject
+    public EurekaReactorHttpClientFactory(EurekaClient eurekaClient, EurekaReverseProxyConfiguration configuration) {
+        this.eurekaClient = eurekaClient;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Optional<HttpClient> newHttpClient(String path) {
```

Comment:
This code can be further condensed to stay within a functional style
```java
return Optional.ofNullable(path)
    .map(this::find)
    .map(Optional::of)
    .orElseGet(() -> resolve(path);
```

Reply:
This should work.  Try it out.  orElseGet() will return an Optional because resolve returns an Optional.  You actually don't need the `map(Optional::of)`.